modules/atmosphere/rendering/atmospheredeferredcaster.cpp

```diff
@@ -103,1085 +103,895 @@ namespace {
+        const glm::vec3& ozoneExtCoeff)
+    {
+        _ozoneExtinctionCoeff = ozoneExtCoeff;
+    }
+
+    void AtmosphereDeferredcaster::setMieScatteringCoefficients(
+        const glm::vec3& mieScattCoeff)
+    {
+        _mieScatteringCoeff = mieScattCoeff;
+    }
+
+    void AtmosphereDeferredcaster::setMieExtinctionCoefficients(const glm::vec3& mieExtCoeff)
+    {
+        _mieExtinctionCoeff = mieExtCoeff;
+    }
+
+    void AtmosphereDeferredcaster::setEllipsoidRadii(const glm::dvec3& radii) {
+        _ellipsoidRadii = radii;
+    }
+
+    void AtmosphereDeferredcaster::setHardShadows(bool enabled) {
+        _hardShadowsEnabled = enabled;
+    }
+
+    void AtmosphereDeferredcaster::setShadowConfigArray(
+        const std::vector<ShadowConfiguration>& shadowConfigArray)
+    {
+        _shadowConfArray = shadowConfigArray;
+    }
+
+    void AtmosphereDeferredcaster::enableSunFollowing(bool enable) {
+        _sunFollowingCameraEnabled = enable;
+    }
+
+    void AtmosphereDeferredcaster::setPrecalculationTextureScale(
+        float preCalculatedTexturesScale)
+    {
+        _calculationTextureScale = preCalculatedTexturesScale;
+        _transmittance_table_width *= static_cast<unsigned int>(_calculationTextureScale);
+        _transmittance_table_height *= static_cast<unsigned int>(_calculationTextureScale);
+        _irradiance_table_width *= static_cast<unsigned int>(_calculationTextureScale);
+        _irradiance_table_height *= static_cast<unsigned int>(_calculationTextureScale);
+        _delta_e_table_width *= static_cast<unsigned int>(_calculationTextureScale);
+        _delta_e_table_height *= static_cast<unsigned int>(_calculationTextureScale);
+        _r_samples *= static_cast<unsigned int>(_calculationTextureScale);
+        _mu_samples *= static_cast<unsigned int>(_calculationTextureScale);
+        _mu_s_samples *= static_cast<unsigned int>(_calculationTextureScale);
+        _nu_samples *= static_cast<unsigned int>(_calculationTextureScale);
+    }
+
+    void AtmosphereDeferredcaster::enablePrecalculationTexturesSaving() {
+        _saveCalculationTextures = true;
+    }
+
+    void AtmosphereDeferredcaster::loadComputationPrograms() {
+        //============== Transmittance T =================
+        if (!_transmittanceProgramObject) {
+            _transmittanceProgramObject = ghoul::opengl::ProgramObject::Build(
+                "transmittanceCalcProgram",
+                absPath("${MODULE_ATMOSPHERE}/shaders/transmittance_calc_vs.glsl"),
+                absPath("${MODULE_ATMOSPHERE}/shaders/transmittance_calc_fs.glsl"));
+        }
+        using IgnoreError = ghoul::opengl::ProgramObject::IgnoreError;
+        _transmittanceProgramObject->setIgnoreSubroutineUniformLocationError(
+            IgnoreError::Yes
         );
-        checkFrameBufferState("_deltaJTableTexture");
-        glViewport(0, 0, _mu_s_samples * _nu_samples, _mu_samples);
-        
```

Comment:
`);` into new line

Reply:
The semi column on line 1026 is in column 72. Are you right about this?

Reply:
Are you talking about the ); from line 1024?